### PR TITLE
Remove dependency on http package

### LINKED ISSUE #20: Don't use the http package
http depends on mirrors, which means that dart2js can't use it. We only use it for http.get, which can be written using just HttpClient without mirrors.

### LINKED ISSUE #1509: Don't use the http package
http depends on mirrors, which means that dart2js can't use it. We only use it for http.get, which can be written using just HttpClient without mirrors.

### PR DESCRIPTION
**Pull request by harryterkelsen**
Fixes dart-lang/tools#1509 
cc @pq @sethladd

### CODE REVIEW

File: lib/discovery.dart [hunk 1]
```diff
@@ -152,10 +152,30 @@ Future<Packages> findPackagesFromNonFile(Uri nonFileUri,
...
 
-/// Fetches a file using the http library.
+/// Fetches a file over http.
 Future<List<int>> _httpGet(Uri uri) {
-  return http.get(uri).then((http.Response response) {
-    if (response.statusCode == 200) return response.bodyBytes;
-    throw 0;  // The error message isn't being used for anything.
+  HttpClient client = new HttpClient();
+  return client.getUrl(uri)
+  .then((HttpClientRequest request) => request.close())
```
Comment (pq) [inline lib/discovery.dart]: Nit: this indentation looks odd...  Are you running the formatter?

Comment (harryterkelsen) [inline lib/discovery.dart]: No, I copied it from https://github.com/dart-lang/sdk/blob/master/pkg/compiler/lib/src/source_file_provider.dart#L71
I'll format this file

File: lib/discovery.dart [hunk 2]
```diff
@@ -5,9 +5,9 @@
 library package_config.discovery;
 
 import "dart:async";
-import "dart:io" show Directory, File, FileSystemEntity;
+import "dart:io";
+import "dart:typed_data" show Uint8List;
 import "package:path/path.dart" as path;
-import "package:http/http.dart" as http;
 import "packages.dart";
 import "packages_file.dart" as pkgfile show parse;
```
Comment (pq) [inline lib/discovery.dart:14]: Nit: while we're here, consider adding a newline separator here as per the [style guide recs](https://www.dartlang.org/articles/style-guide/#ordering).

Comment (harryterkelsen) [inline lib/discovery.dart:14]: done

Comment (pq) [inline lib/discovery.dart:14]: Thanks!

**Comment by pq** [comment]
Forgot to say: LGTM!

**Pull request closed without merging**